File: common/src/main/java/mconi/common/sim/OniWorldFoundation.java
package mconi.common.sim;

/**
 * Query helpers for constrained-world topology rules.
 */
public final class OniWorldFoundation
{
	private OniWorldFoundation() { }

	public static boolean isWithinHorizontalBounds(int x, int z, OniSimulationConfig config)
	{
		return x >= config.worldMinX() && x <= config.worldMaxX()
				&& z >= config.worldMinZ() && z <= config.worldMaxZ();
	}

	public static boolean isVoidBand(int y, int maxY, OniSimulationConfig config)
	{
		return y >= (maxY - config.voidBandHeight() + 1);
	}

	public static boolean isLavaBand(int y, int minY, OniSimulationConfig config)
	{
		return y <= (minY + config.lavaBandHeight() - 1);
	}
}

File: common/src/main/kotlin/mconi/common/sim/OniWorldFoundation.kt
package mconi.common.sim

/**
 * Query helpers for constrained-world topology rules.
 */
object OniWorldFoundation {
    @JvmStatic
    fun isWithinHorizontalBounds(x: Int, z: Int, config: OniSimulationConfig): Boolean {
        return x >= config.worldMinX() && x <= config.worldMaxX() &&
            z >= config.worldMinZ() && z <= config.worldMaxZ()
    }

    @JvmStatic
    fun isVoidBand(y: Int, maxY: Int, config: OniSimulationConfig): Boolean {
        return y >= (maxY - config.voidBandHeight() + 1)
    }

    @JvmStatic
    fun isLavaBand(y: Int, minY: Int, config: OniSimulationConfig): Boolean {
        return y <= (minY + config.lavaBandHeight() - 1)
    }
}
